src/app/Library/CrudPanel/CrudField.php

```diff
@@ -231,6 +231,22 @@ public function validationMessages(array $messages)
         return $this;
     }
 
+    public function entity(string $entity)
+    {
+        $this->attributes['entity'] = $entity;
+
+        if ($entity !== false) {
+            $this->attributes = $this->crud()->makeSureFieldHasRelationType($this->attributes);
+            $this->attributes = $this->crud()->makeSureFieldHasModel($this->attributes);
+            $this->attributes = $this->crud()->makeSureFieldHasAttribute($this->attributes);
+            $this->attributes = $this->crud()->makeSureFieldHasMultiple($this->attributes);
+            $this->attributes = $this->crud()->makeSureFieldHasPivot($this->attributes);
+            $this->attributes = $this->crud()->makeSureFieldHasType($this->attributes);
+        }
```

Comment:
Can't we create a `public` method that calls all of these potentially `private` methods? Say... `makeSureFieldHasAllRelationshipAttributes()`? Would that be a good idea to keep those methods `protected`?